core/src/main/java/io/questdb/cairo/wal/WalUtils.java

```diff
@@ -96,4 +106,76 @@ public static void createTxnLogFile(FilesFacade ff, MemoryMARW mem, Path txnSeqD
             txnSeqDirPath.trimTo(rootLen);
         }
     }
+
+    /**
+     * Retrieves the last refresh state for a materialized view.
+     * <p>
+     * Note: This function is intended to be used during the initialization phase only.
+     * It has been extracted into `WalUtils` solely to facilitate its usage in tests.
+     *
+     * @param tablePath          the path to the table
+     * @param tableToken         the table token
+     * @param configuration      the Cairo configuration
+     * @param txnLogMemory       the memory to iterate over transaction logs
+     * @param walEventReader     the reader to read WAL events
+     * @param blockFileReader    the reader to read the state file
+     * @param matViewStateReader the reader used to collect the state
+     * @return true if the last mat view state was successfully retrieved, false otherwise
+     */
+    public static boolean readMatViewState(
+            Path tablePath,
+            TableToken tableToken,
+            CairoConfiguration configuration,
+            MemoryCMR txnLogMemory,
+            WalEventReader walEventReader,
+            BlockFileReader blockFileReader,
+            MatViewStateReader matViewStateReader
+    ) {
+        try (MemoryCMR mem = txnLogMemory) {
+            final int tablePathLen = tablePath.size();
+            mem.smallFile(configuration.getFilesFacade(), tablePath.concat(SEQ_DIR).concat(TXNLOG_FILE_NAME).$(), MemoryTag.MMAP_TX_LOG);
+            final long txnCount = mem.getLong(TableTransactionLogFile.MAX_TXN_OFFSET_64);
+            if (txnCount > 0) {
+                final long fileSize = TableTransactionLogFile.HEADER_SIZE + txnCount * TableTransactionLogV1.RECORD_SIZE;
```

Comment:
We need to check log version and, if it's V2, use `TableTransactionLogV2`. We need some tests around that.

Reply:
fixed in: https://github.com/questdb/questdb/pull/5519/commits/4014ee8b06a1b879f32671993e9e566a14e2e43f

Reply:
`testReadMatViewStateInvalidFileFormat` seems to be failing on Windows